server/Episememe.Application.Tests/FeatureTests/FileUpload/WhenUploadingFile.cs

```diff
@@ -202,12 +208,51 @@ public void GivenMediaWithExistingTag_ExistingTagIsUsed()
 
             var command = FileUploadCommand.Create(formFile, givenMediaDto, string.Empty);
             var handler = new FileUploadCommandHandler(_fileStorageMock, _contextMock,
-                _mediaIdProviderMock.Object);
+                _mediaIdProviderMock.Object, _timeProviderMock.Object);
             handler.Handle(command, CancellationToken.None).Wait();
 
             _contextMock.Tags.Should().ContainSingle(t => t.Name == givenTag);
         }
 
+        [Fact]
+        public void GivenNewMediaIsUploaded_MediaChangeIsSavedInDatabase()
+        {
+            var givenFileName = "newFile";
+            var givenFileContent = "None";
+            var givenFormFile = FormFileFactory.Create(givenFileName, givenFileContent);
+            var givenTags = new[] { "pigeons", "flying rats" };
+            var givenMediaDto = new FileUploadDto(givenTags, _givenDate, false);
+            var givenUserId = "user";
+
+            var command = FileUploadCommand.Create(givenFormFile, givenMediaDto, givenUserId);
+            var handler = new FileUploadCommandHandler(_fileStorageMock, _contextMock,
+                _mediaIdProviderMock.Object, _timeProviderMock.Object);
+            handler.Handle(command, CancellationToken.None).Wait();
+
+            _contextMock.MediaChanges.Should().ContainSingle(mc => mc.MediaInstanceId == _givenMediaInstanceId
+                                                             && mc.UserId == givenUserId);
+        }
+
+        [Fact]
+        public void GivenMediaChangeIsSavedInDatabase_MediaChangeIsOfTypeCreate()
+        {
+            var givenFileName = "newFile";
+            var givenFileContent = "None";
+            var givenFormFile = FormFileFactory.Create(givenFileName, givenFileContent);
```

Comment:
File name and file content are not really that important in this test, you can just use string literals in Create()